src/libraries/Directory.Build.props

```diff
@@ -304,9 +308,11 @@
 
   <PropertyGroup Condition="'$(BuildAllConfigurations)' != 'true'">
     <!-- We add extra binplacing for the test shared framework until we can get hardlinking with the runtime directory working on all platforms -->
-    <BinPlaceTestSharedFramework Condition="'$(BuildTargetFramework)' == '$(NetCoreAppCurrent)'">true</BinPlaceTestSharedFramework>
+    <BinPlaceTestSharedFramework Condition="'$(BuildTargetFramework)' == '$(NetCoreAppCurrent)' and '$(TargetsMobile)' != 'true'">true</BinPlaceTestSharedFramework>
```

Comment:
So then this works fine?

Reply:
I just ran locally, and it was still producing the files that were excluded previously in `runtime.depproj`. It still built fine, but we might need to reintroduce this.
```
<!-- In-tree runtime packs do not require the host, so exclude --> 	
  <ItemGroup Condition="'$(BinPlaceTestRuntimePack)' == 'true'">	
    <FileToExclude Include="dotnet" />	
    <FileToExclude Include="libnethost" />	
    <FileToExclude Include="libhostfxr" />	
    <FileToExclude Include="libhostpolicy" />	
    <FileToExclude Include="nethost" />	
  </ItemGroup>
```

Reply:
Yeah that is because of the `PackageReferences` as I explained here: https://github.com/dotnet/runtime/pull/34662#discussion_r407808321

@ViktorHofer I think we should condition the host package references for this scenario, or would you rather have the `FileToExclude`?

Reply:
Conditioning the package refs seems like the better solution. We want to remove the runtime.depproj in the (near) future, so not using depproj filter logic is good :)

Reply:
Yeah I agree.

Reply:
Done, added conditions to a few of the package refs.